### PR TITLE
refactor(webapp): captcha to component 

### PR DESCRIPTION
**Pull request by Rotzbua**
Refactor redundant captcha code to one component.

## Reference

 #752

### CODE REVIEW

**Comment by Rotzbua** [comment]
@peterthomassen First working draft for captcha. The proxy commit is just for testing and will be removed.

**Comment by peterthomassen** [comment]
Display works fine. Submitting the captcha returns a 400 response with payload

```
{
    "captcha": {
        "id": [
            "This field is required."
        ],
        "solution": [
            "This field is required."
        ]
    }
}
```

DOM isn't updated after the 400 response (i.e. spinner moves indefinitely on submission button). Console says:
```
[Vue warn]: Error in callback for watcher "captcha_error": "TypeError: Cannot read properties of undefined (reading 'getCaptcha')"

found in

---> <ActivateAccountActionHandler> at /usr/src/app/src/components/ActivateAccountActionHandler.vue
       <VForm>
         <VCard>
           <ConfirmationPage> at /usr/src/app/src/views/ConfirmationPage.vue
             <VMain>
               <VApp>
                 <App> at /usr/src/app/src/App.vue
                   <Root>
warn @ chunk-TQVERUCE.js?v=08bf4eec:3674


TypeError: Cannot read properties of undefined (reading 'getCaptcha')
    at VueComponent.captcha_error (ActivateAccountActionHandler.vue:52:1)
    at invokeWithErrorHandling (chunk-TQVERUCE.js?v=08bf4eec:2396:26)
    at Watcher2.run (chunk-TQVERUCE.js?v=08bf4eec:2788:13)
    at flushSchedulerQueue (chunk-TQVERUCE.js?v=08bf4eec:3271:13)
    at Array.<anonymous> (chunk-TQVERUCE.js?v=08bf4eec:2481:12)
    at flushCallbacks (chunk-TQVERUCE.js?v=08bf4eec:2438:14)


[Vue warn]: Error in v-on handler (Promise/async): "TypeError: Spread syntax requires ...iterable[Symbol.iterator] to be a function"

found in

---> <VForm>
       <VCard>
         <ConfirmationPage> at /usr/src/app/src/views/ConfirmationPage.vue
           <VMain>
             <VApp>
               <App> at /usr/src/app/src/App.vue
                 <Root>
```

**Comment by peterthomassen** [comment]
Would you mind rebasing this? Then I'll take a another look.

File: www/webapp/src/components/Field/GenericCaptcha.vue [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    inputSolution: '',
+    rules: [v => !!v || 'Please enter the text displayed in the picture so we are (somewhat) convinced you are human.'],
+    errors: [],
+    kind: 'image',
+    mimeAudio: 'audio/wav',
+    mimeImage: 'image/png',
+    l: {
+      altImage: 'Sign up / password reset is also possible by sending an email to our support.',
+      hintAudio: 'Can\'t see? Hear an audio CAPTCHA instead.',
+      hintImage: 'Trouble hearing? Switch to an image CAPTCHA.',
```
Comment (peterthomassen) [inline www/webapp/src/components/Field/GenericCaptcha.vue:93]: Those two strings are swapped.

File: www/webapp/src/views/ResetPassword.vue [hunk 2]
```diff
@@ -52,53 +52,11 @@
...
-                                    &nbsp;
-                                    <v-btn-toggle v-model="captcha_kind">
-                                      <v-btn text outlined value="image" aria-label="Switch to Image CAPTCHA" :disabled="captchaWorking"><v-icon>{{ mdiEye }}</v-icon></v-btn>
-                                      <v-btn text outlined value="audio" aria-label="Switch to Audio CAPTCHA" :disabled="captchaWorking"><v-icon>{{ mdiEarHearing }}</v-icon></v-btn>
-                                    </v-btn-toggle>
-                                  </v-col>
-                                </v-row>
-                              </v-container>
+                            <generic-captcha
+                                @update="(id, solution) => {captchaID=id; captchaSolution=solution}"
```
Comment (peterthomassen) [inline www/webapp/src/views/ResetPassword.vue:56]: nice

**Review by peterthomassen — CHANGES_REQUESTED** [review]
Almost ready! Just two strings have been confused.

**Review by peterthomassen — APPROVED** [review]

**Pull request merged**